pgjdbc/src/main/java/org/postgresql/util/PGServiceFile.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2018, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.text.MessageFormat;
+import java.util.HashMap;
+
+/*
+ * See https://www.postgresql.org/docs/current/static/libpq-pgservice.html.
+ */
+public class PGServiceFile {
+  public static HashMap load(String service) throws Exception {
+    String filename = findPath();
+    PGServiceFile file = new PGServiceFile();
+    BufferedReader in = new BufferedReader(new FileReader(filename));
```

Comment:
There should be file encoding somewhere. The default one is not reliable.

Reply:
@vlsi ok, where should i find the correct encoding ? Which API do you suggest me ?

Reply:
The question goes to `hackers` team :)
What is the encoding of `pg_service.conf` file?
Frankly speaking, I've no idea.

Reply:
I switched to `Scanner` API and `US-ASCII` encoding.